core/systems/disks/steam_removable_media_manager.gd

```diff
@@ -0,0 +1,241 @@
+const STEAMOS_TRIM_DEVICES = "/usr/bin/steamos-polkit-helpers/steamos-trim-devices"
+const BLOCK_PREFIX = "/org/freedesktop/UDisks2/block_devices"
+const SDCARD_PATH = "/dev/mmcblk0"
+
+const protected_mounts = [
+	"/",
+	"/boot",
+	"/boot/efi",
+	"/efi",
+	"/frzr_root",
+	"/frzr_root/boot",
+	"/home",
+	"/var",
+	"/var/cache",
+	"/var/log",
+	]
+
+var format_capable: bool = false
+var init_capable: bool = false
+var format_sd_capable: bool = false
+var retrigger_capable: bool = false
+var trim_capable: bool = false
+
+
+signal drives_updated(device_tree: Array[BlockDevice])
+
+## Hierarchy of drive -> block device -> partitions -> mount points
+## Used to display device information to the user.
+var device_tree: Array[BlockDevice]
+
+func _init() -> void:
+
+	# Check the required system files exist for steam_removable_media
+	if FileAccess.file_exists(SRM_FORMAT_MEDIA):
+		format_capable = true
+	if FileAccess.file_exists(SRM_INIT_MEDIA):
+		init_capable = true
+	if FileAccess.file_exists(STEAMOS_RETRIGGER_AUTOMOUNTS):
+		retrigger_capable = true
+
+	# These scripts are hard coded to mmcblk0
+	if FileAccess.file_exists(SDCARD_PATH):
+		if FileAccess.file_exists(STEAMOS_FORMAT_SDCARD):
+			format_sd_capable = true
+		if FileAccess.file_exists(STEAMOS_TRIM_DEVICES):
+			trim_capable = true
+
+	logger.debug("format_capable: " + str(format_capable))
+	logger.debug("init_capable: " + str(init_capable))
+	logger.debug("format_sd_capable: " + str(format_sd_capable))
+	logger.debug("retrigger_capable: " + str(retrigger_capable))
+	logger.debug("trim_capable: " + str(trim_capable))
+
+	# Get the current drives and ensure they are always up to date
+	udisks2.devices_updated.connect(update_drives)
+	update_drives()
+
+
+## Rebuilds the device_tree with the current device list
+func update_drives(devices: Array[BlockDevice] = []):
+	if devices == []:
+		logger.debug("Got empty array, grabbing devices from udisks2")
+		devices = udisks2.get_devices()
+	self.device_tree = []
+	for block in devices:
+		if not _block_device_has_protected_mount(block):
+			logger.debug(block.dbus_path, "has no protected mounts.")
+			device_tree.append(block)
+	self.drives_updated.emit(self.device_tree)
+
+
+## Calls the SteamRemovableMedia format-media script to format a drive as EXT4
+## and intialize it as a steam library
+func format_drive(device: BlockDevice) -> void:
+	if not format_capable:
+		logger.error("System is not format capable")
+		return
+
+	# This should never hit if using our device tree, but  the method is public so
+	# make sure.
+	if _block_device_has_protected_mount(device):
+		logger.error("Attempted to format device with protected mount. Illegal Operation.")
+		return
+
+	var drive = "/dev" + device.dbus_path.trim_prefix(BLOCK_PREFIX)
+	logger.debug("Formatting drive: " + drive)
+	var args := ["--full", "--device", drive]
+	var result = await _execute_in_thread(SRM_FORMAT_MEDIA, args)
+	logger.debug("Got results: " + str(result[0]), str(result[1]))
```

Comment:
FYI, with the logger, you can now simply do this:
```suggestion
	logger.debug("Got results:", result[0], result[1])
```
Using this new way is actually preferred, because it will skip doing any string conversion if the log level doesn't match, which will save us some CPU cycles.